### DISCUSSION TITLE
guide-R/manipulation/etiquettes-valeurs

### DISCUSSION THREAD

**Discussion by giscus** [larmarange/guide-R#36] [Forum]
# guide-R/manipulation/etiquettes-valeurs

https://larmarange.github.io/guide-R/manipulation/etiquettes-valeurs.html

**Comment by stephanesq** [comment]
Bonjour, je me suis posé la question lors de la production de tableaux pour une analyse avec des catégories un peu longues : est-ce qu'il vaut mieux attribuer des étiquettes de valeurs ou coder en dur les facteurs ?
Je n'ai pas trouvé de réponse satisfaisante jusqu'ici.
Merci encore pour la doc.

**Reply by larmarange** [reply]
Bonjour,

en termes de stockage mémoire, cela ne change pas grand chose. La différence entre facteurs et vecteurs numériques labellisées est avant tout de pouvoir conserver le codage d'origine de l'enquête.

Par contre, au moment de réaliser des statistiques (statistiques descriptives, modèles), les variables catégorielles doivent impérativement être converties en facteurs pour être traitées correctement. Si vous aviez conservé des étiquettes de valeurs, vous pouvez procéder aux conversions avec `to_factor()` ou `unlabelled()`.

Pour produire des graphiques / tableaux propres en vue d'une publication, il est souvent, dans les deux cas, nécessaire d'ajuster les étiquettes pour qu'elles soient explicites et pertinentes.

Cdlt
